### PR TITLE
Add aggregate function for multi-kernel transaction

### PR DESCRIPTION
**Pull request by quentinlesceller**
This PR adds an ```aggregate``` function into ```core/transaction.rs```. This function will aggregate multiple transactions with cut-through.
While unused at the moment, this function will be useful for potential transaction aggregation in the stempool.
See https://github.com/mimblewimble/grin/issues/819.

### CODE REVIEW

**Comment by tromp** [comment]
The following test cases should be added, because we may want to change the behaviour of these edge cases, like for example indicate an error.
1. aggregating a coinbase output with a spend of that coinbase.
2. aggregating the three txs {a->b, b->c, b->d}
currently 1. produces a tx with overlap between inputs and outputs, while 2 produces in invalid tx a -> c,d

**Comment by quentinlesceller** [comment]
Thanks for the review. 
1. Currently Coinbase outputs are filtered out as you can see [here](https://github.com/mimblewimble/grin/pull/966/files#diff-404c77cf79ebda67cdbd58c1c99daa8cR511). So the aggregation will work but will not do anything in that case.
2. As of now, the aggregation function will build a multi-kernel transaction with such transactions. However, this transaction will be rejected by the pool. In the future, the aggregation function will work in conjunction with the stempool (in that case one of the last 2 transaction will get rejected). I will add a test to ensure that it is the case.

File: core/src/core/transaction.rs [hunk 1]
```diff
@@ -461,6 +462,75 @@ impl Transaction {
...
+		.cloned()
+		.collect::<Vec<_>>();
+
+	let new_outputs = outputs
+		.iter()
+		.filter(|out| !to_cut_through.contains(&out.commitment()))
+		.cloned()
+		.collect::<Vec<_>>();
+
+	let tx = Transaction::new(new_inputs, new_outputs, kernels);
```
Comment (antiochp) [inline core/src/core/transaction.rs]: We need to sort the `inputs|outputs|kernels` here to ensure the tx is valid.
And we should cover this in the test for completeness.

Comment (quentinlesceller) [reply core/src/core/transaction.rs]: I just added the sorting and the test in ```core/mod.rs```.
